llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -1371,7 +1378,12 @@ void GCNPassConfig::addPreRegBankSelect() {
 }
 
 bool GCNPassConfig::addRegBankSelect() {
-  addPass(new AMDGPURegBankSelect());
+  if (NewRegBankSelect) {
+    addPass(createAMDGPURBSelectPass());
+    addPass(createAMDGPURBLegalizePass());
+  } else {
+    addPass(new AMDGPURegBankSelect());
```

Comment:
Could we just modify this AMDGPURegBankSelect instead of adding a new one? It was meant to use uniformity analysis too. Maybe you can propagate the flag into the pass if you want to keep the old behaviour around in the meantime.

Reply:
Older versions were doing that. New passes are much simpler and don't use anything from generic reg-bank-select(except register banks that come from td file via RegisterBankInfo).
To simplify review process and transition period I would prefer two passes separate from AMDGPURegBankSelect, rb-legalize will also be split across few files.
Old behavior is there to keep upcoming patches small by gradually switching to -new-reg-bank-select.